specs/PeriodicBackgoundSyncAndBackgroundSync.md

```diff
@@ -0,0 +1,581 @@
+===
+
+# Background
+There're WebView2 customers who want to run tasks in the background at 
+periodic intervals for purposes such as updating code, caching data, 
+and pre-loading resources. Based on that, the WebView2 team is adding 
+support for [Periodic_Background_Synchronization](https://developer.mozilla.org/docs/Web/API/Web_Periodic_Background_Synchronization_API). 
+In the meanwhile, the WebView2 team is also adding support for 
+[Background_Synchronization](https://developer.mozilla.org/docs/Web/API/Background_Synchronization_API)
+which enhances the offline experience for web applications.
+
+Unlike [Progressive Web App](https://learn.microsoft.com/microsoft-edge/progressive-web-apps-chromium/how-to/background-syncs#use-the-periodic-background-sync-api-to-regularly-get-fresh-content)
+which passively run periodic background tasks based on the browser's 
+heuristic, WebView2 exposes an API to give developers complete control 
+over how to run a periodic background synchronization task.
+
+# Description
+We propose following APIs:
+
+**SyncRegistrationManager API**: This API gives developers the ability 
+to manage the periodic background synchronizations and background sync 
+synchronizations in that service worker. 
+
+**PeriodicSyncRegistered API**: This API gives developers the ability 
+to subscribe to the event when a new Periodic Background synchronization 
+task is registered.
+
+**BackgroundSyncRegistered API**: This API gives developers the ability 
+to subscribe to the event when a new Background synchronization task is 
+registered.
+
+**GetSyncRegistrations API**: This asynchronous call returns the 
+collection of all periodic background synchronization registrations or 
+background synchronization registrations based on the 
+`CoreWebView2ServiceWorkerSynchronizationKind` provided. We add
+`CoreWebView2ServiceWorkerSyncRegistrationInfo` with `Tag` and 
+`MinInterval` properties to represent a synchronization registration.
+
+**DispatchPeriodicSyncEvent API**: This API gives developers the ability 
+to trigger a periodic background synchronization task based on the 
+tag name. Developers can use native OS timer to implement a scheduler 
+to trigger periodic synchronization task according to their own logic.
+
+# API Details
+## C++
+```
+/// Indicates the service worker background synchronization type.
+[v1_enum]
+typedef enum COREWEBVIEW2_SERVICE_WORKER_SYNCHRONIZATION_KIND {
+  /// Indicates that the synchronization is a background synchronization.
+  /// See [Background Synchronization](https://developer.mozilla.org/docs/Web/API/Background_Synchronization_API)
+  /// for more information.
+  COREWEBVIEW2_SERVICE_WORKER_SYNCHRONIZATION_KIND_BACKGROUND_SYNC,
+  /// Indicates that the synchronization is a periodic background synchronization.
+  /// See [Periodic Background Synchronization](https://developer.mozilla.org/docs/Web/API/Web_Periodic_Background_Synchronization_API)

```

Comment:
I like this link to the MDN doc on the two kinds of syncs. Please also add more links like this in the ref docs below for the Tag property, and the events or anywhere else where the thing in our WebView2 API directly corresponds to something in the DOM APIs.